Framework/DataObjects/src/ScanningWorkspaceBuilder.cpp

```diff
@@ -194,10 +195,14 @@ MatrixWorkspace_sptr ScanningWorkspaceBuilder::buildWorkspace() const {
   auto outputWorkspace = create<Workspace2D>(
       m_instrument, m_nDetectors * m_nTimeIndexes, m_histogram);
 
-  auto &outputDetectorInfo = outputWorkspace->mutableDetectorInfo();
-  outputDetectorInfo.setScanInterval(m_timeRanges[0]);
+  auto &outputComponentInfo = outputWorkspace->mutableComponentInfo();
+  outputComponentInfo.setScanInterval(
+      {m_timeRanges[0].first.totalNanoseconds(),
+       m_timeRanges[0].second.totalNanoseconds()});
```

Comment:
You still need to make the API consistent here and use `DateAndTime` instead of nanoseconds.